### ISSUE TITLE
Botania Petal Apothacary not accepting custom recipes.

### ISSUE THREAD

**Issue by goosezilla** [jaredlll08/ModTweaker#396]
Same NBT issue? Dont see how this one can be related to JEI though as it doesnt return any flower with that recipe.

Script: http://hastebin.com/acasuzizoh.xml
Works with an output that isnt NBT tagged.

Recipe showing in JEI: http://i.imgur.com/DxdeQ6m.png

Apothecary not accepting custom recipe items: http://i.imgur.com/QdZysAq.png

**Comment by jaredlll08** [comment]
Once again. you are using the wrong thing!
http://minetweaker3.powerofbytes.com/wiki/Tutorial:Advanced_Recipes
please read the NBT stuff there.

**Comment by goosezilla** [comment]
Provide me with ANY working recipe that returns an item with NBT. None of those work. Please be more helpful than "read the NBT stuff".

**Comment by jaredlll08** [comment]
```
val pick = <minecraft:stone_pickaxe>;
val stonedPick = pick.withTag({display: {Name: "Stoned pick", Lore: ["This pick", "Has been experimenting too much"]}});
val cobble = <minecraft:cobblestone>;
recipes.addShaped(stonedPick, [[pick, cobble], [cobble, cobble]]);
```
When helping, I assume that you have read the wiki, that is why I replied in that way, if you have not read the wiki, then do so.

**Comment by goosezilla** [comment]
Doesnt that example prove my point that NBT is not being used correctly? The input pick can be the stonedpick and the recipe still works.

I really meant a petal apothecary recipe that works anyway. Ive spent about 5 hours trying every combination from the wiki with the apothecary and none of them work. I have a co pack developer who also cant get NBT to work properly.

**Comment by goosezilla** [comment]
Am I wrong? Seriously if its so obvious and easy can you not just help me out and fix the recipe in the script above? Ive spent so long trying to get this to work using the wiki and im only here because I cant get it to work after trying everything I can think of. 

Sorry if this is some massive inconvenience to you but I really dont see an answer and the example provided above is not helpful for this situation. Ive tried the EXACT same code for the petal apothecary and cant get it to give me a valid recipe that accepts the input items. 

Please dont just tell me im doing it wrong without providing any specific help or example other than to go back to doing the equivalent of banging my head against a brick wall.

**Comment by Lanse505** [comment]
@goosezilla may I see your script?

**Comment by goosezilla** [comment]
http://hastebin.com/acasuzizoh.xml

theres that one but i mean I feel like ive tried every possible variation of the values and withTag/onlyWithTag so its just one variation of the script that I really feel should work.

**Comment by Lanse505** [comment]
Not at my main pc atm but could you try this?

```
//Botania - Endoflame
var endoflame = <botania:specialFlower>.withOnlyTag({type:"endoflame"});

mods.botania.Apothecary.removeRecipe(endoflame);
mods.botania.Apothecary.addRecipe(endoflame, [<ore:petalBrown>, <ore:petalBrown>, <ore:petalRed>, <ore:petalLightGray>, <minecraft:wool>]);
```

**Comment by goosezilla** [comment]
tried it and its the exact same problem

**Comment by Lanse505** [comment]
@goosezilla 

Just did some testing.
This is not an issue with NBT (Since I double checked it against the proper way of removing and adding flower recipes via the API.

Aka
//Botania - Endoflame

mods.botania.Apothecary.removeRecipe("endoflame");
mods.botania.Apothecary.addRecipe("endoflame", [<ore:petalBrown>, <ore:petalBrown>, <ore:petalRed>, <ore:petalLightGray>, <minecraft:wool>]);

**Comment by Lanse505** [comment]
This as per:

```
    @ZenMethod
    public static void addRecipe(String output, IIngredient[] input) {
        addRecipe(toIItemStack(ItemBlockSpecialFlower.ofType(output)), input);
    }

    private static class Add extends BaseListAddition<RecipePetals> {
        public Add(RecipePetals recipe) {
            super("Botania Petal", BotaniaAPI.petalRecipes);
            recipes.add(recipe);
        }
```

&


```
    @ZenMethod
    public static void removeRecipe(String output) {
        removeRecipe(toIItemStack(ItemBlockSpecialFlower.ofType(output)));
    }
```

**Comment by Lanse505** [comment]
However the recipe is still broken which points towards an issue with the Botania/Minetweaker integration that still needs to be investigated.

TLDR;
Not NBT, It's a broken Integration.
Will look into.

**Comment by Lanse505** [comment]
Me and Jared are troubleshooting the issue with the integration to see if it's CrT's Issue or Botanias Issue.

**Comment by goosezilla** [comment]
Thank you.
